src/prefect/server/api/concurrency_limits_v2.py

```diff
@@ -266,3 +296,71 @@ async def bulk_decrement_active_slots(
         )
         for limit in limits
     ]
+
+
+ONE_HOUR = 60 * 60
+
+
+@router.post("/active_owners", status_code=status.HTTP_200_OK)
+async def read_concurrency_limit_owners(
+    names: List[str] = Body(..., min_items=1),
+    time_window: int = Body(ONE_HOUR, le=ONE_HOUR),
+    db: PrefectDBInterface = Depends(provide_database_interface),
+) -> Dict[str, Any]:
+    async with db.session_context(begin_transaction=True) as session:
+        limits = (
+            await models.concurrency_limits_v2.bulk_read_or_create_concurrency_limits(
+                session=session, names=names, create_if_missing=False
+            )
+        )
+
+        if not limits:
+            return {}
+
+    # Create an EventFilter to find all concurrency limit acquired and released events for
+    # resources matching the concurrency limit names in `limits`
+    since = datetime.now() - timedelta(seconds=time_window)
+
+    limit_ids = [f"prefect.concurrency-limit.{limit.id}" for limit in limits]
+    acquired_filter = EventFilter(
+        occurred={"since": since},
+        event={"name": ["prefect.concurrency-limit.acquired"]},
+        resource={"id": limit_ids},
+    )
+    released_filter = EventFilter(
+        occurred={"since": since},
+        event={"name": ["prefect.concurrency-limit.released"]},
+        resource={"id": limit_ids},
+    )
+
+    acquired_events = await get_all_events(acquired_filter, db)
+    released_events = await get_all_events(released_filter, db)
```

Comment:
Might be more efficient to ask for them all in one query?  `event={"name": ["prefect.concurrency-limit.acquired", "prefect.concurrency-limit.released"]}` would do that.  Then you'd just need to filter on the event names in your loops below

Reply:
Or another way to keep it cleaner would be to have the `get_all_events` function split them out for you, so it looked like:

```
acquired_events, released_events = await get_all_events(filter, db)
```

Reply:
Oh, I like that one.